source/utilities.cc

```diff
@@ -1338,7 +1348,10 @@ namespace aspect
           const unsigned int column_num = field_index%(components+dim);
 
           if (column_num >= dim)
-            temp_data *= scale_factor;
+            {
+              temp_data *= scale_factor;
+              maximum_component_value[column_num-dim] = std::max(maximum_component_value[column_num-dim], temp_data);
```

Comment:
Same here, a max between 0 and the data.

Reply:
I added it to the resize function.